### PR TITLE
NN-2935 new pagination service

### PR DESCRIPTION
**Pull request by JamieDraperUK**
image

image

image

### CODE REVIEW

File: backend/controllers/prisonerProfile/adjudicationHistory.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ module.exports = ({ adjudicationHistoryService, elite2Api, logError, paginationS
             result.adjudicationNumber
           } </a>`,
         },
-        { text: `${result.reportDate} ${result.reportTime}` },
+        { text: `${result.reportDate} - ${result.reportTime}` },
```
Comment (Nimphal) [inline backend/controllers/prisonerProfile/adjudicationHistory.js:75]: Sneaky :D

File: package.json [hunk 2]
```diff
@@ -171,7 +171,7 @@
   },
   "scripts": {
     "start": "node backend/index.js",
-    "start:dev": "nodemon --watch backend --watch views backend/index.js",
+    "start:dev": "nodemon --watch backend --watch views backend/index.js | bunyan -o simple",
```
Comment (Nimphal) [inline package.json:174]: :+1:

File: backend/controllers/prisonerProfile/adjudicationHistory.js (hunk 1 above)
Comment (JamieDraperUK) [reply backend/controllers/prisonerProfile/adjudicationHistory.js:75]: haha. Told Dom I'd sort it in my next PR

File: backend/routes/prisonerProfileRouter.js [hunk 3]
```diff
@@ -18,7 +18,8 @@ const prisonerIncentiveLevelDetails = require('../controllers/prisonerProfile/pr
 
 const prisonerProfileServiceFactory = require('../services/prisonerProfileService')
 const personServiceFactory = require('../services/personService')
-const paginationService = require('../services/paginationService')
+const paginationService = require('../services/pagination2Service')
```
Comment (JamieDraperUK) [inline backend/routes/prisonerProfileRouter.js:21]: pagination2Service is a temp name at the moment as we're not sure if we want to make this the default pagination service.

File: backend/services/pagination2Service.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      text: page + 1,
+      href: url.href,
+      selected: offset === pageOffset,
+    }
+  }
+
+  const useLowestNumber = (left, right) => (left >= right ? right : left)
+
+  const calculateFrom = ({ currentPage, numberOfPages }) => {
+    const towardsTheEnd = numberOfPages - currentPage <= pageBreakPoint
```
Comment (Nimphal) [inline backend/services/pagination2Service.js:67]: As poetic a variable as any.

**Review by Nimphal — APPROVED** [review]
LGTM!

Great work!

File: backend/services/pagination2Service.js (hunk 4 above)
Comment (JamieDraperUK) [reply backend/services/pagination2Service.js:67]: 🤣

**Review by Nimphal — DISMISSED** [review]

**Review by jsrobertson — APPROVED** [review]

**Pull request merged**